### PR TITLE
Use <q-args> instead of <args>

### PR DESCRIPTION
**Pull request by mvanderkamp**
This allows calling commands like this:
  `VimuxRunCommand echo Value in '$HOME': $HOME`
Note that no quotes are needed around the command string!
Wrapping the command in quotation characters is still supported though, but with a deprecation warning message.

### CODE REVIEW

File: plugin/vimux.vim [hunk 1]
```diff
@@ -64,7 +64,32 @@ function! VimuxRunLastCommand()
...
+" Used to make sure that the transition to using <q-args> instead of just
+" <args> is painless.
+function! s:trimQuotes(command)
+  let length = strcharlen(a:command)
+  if length > 0
+    let l:start = strgetchar(a:command, 0)
+    let l:end = strgetchar(a:command, length-1)
+    if l:start == l:end && index(['"', "'"], nr2char(l:start)) >= 0
+      call s:deprecationWarning('No longer necessary to wrap command string in quotes.')
+      echohl None
```
Comment (mvanderkamp) [inline plugin/vimux.vim:84]: Whoops this line shouldn't be here. I'll remove it.

**Comment by mvanderkamp** [comment]
I've also just realized that the documentation suggests calling the commands as though they were functions, but without using `:call`:
```
map <Leader>vm :VimuxPromptCommand("make ")<CR>
```
Which would also not work anymore with `<q-args>` without special handling. I still think we should do this though- I'm guessing this was an oversight, and the fact that this suggestion worked before was probably an accident. I'll patch up this PR.

**Comment by alerque** [comment]
I think I tried doing this already and had to back it out because of how many problems it caused with people's existing configs, and (if memory servers me) some of them were not really adaptable. I'll have to dig up what happened there and review the fallout before I can make a judgement call on this.

**Comment by alerque** [comment]
Apparently my memory is aging badly. The previous snafu wasn't `q-` related, it was `-bar` related. See 1b38911 commit message for reversal.

Do you have any idea how `-bar` commands and `<q-args>` get along?

**Comment by alerque** [comment]
By the way [grep.app](https://grep.app) is quite useful for situations like this to double check [real world usage](https://grep.app/search?q=%3AVimuxPromptCommand). Only a very small minority of people have their vimrc's publicly posted, but I've found the samplings that are posted tend to be representative: both of people that understand and innovate with their configs and people that just copy blobs they find somewhere.

**Comment by mvanderkamp** [comment]
> By the way [grep.app](https://grep.app) is quite useful for situations like this to double check [real world usage](https://grep.app/search?q=%3AVimuxPromptCommand). Only a very small minority of people have their vimrc's publicly posted, but I've found the samplings that are posted tend to be representative: both of people that understand and innovate with their configs and people that just copy blobs they find somewhere.

Very useful, thank you! Sadly yes it does indeed look like there are projects, including fairly popular ones, that use the "command that looks like a call" syntax. I believe with these latest touchups though that they should still work, albeit with a deprecation message. Don't take my word on that though...

As for `-bar`, I think that will mess with the way these commands are current invoked, but that it plays relatively nicely with `<q-args>`, so moving to `<q-args>` and deprecating wrapping the command arguments in strings should be a step towards support `-bar` if we want to go there. It's a bit iffy though since it makes using quotes in the command string difficult. And honestly since we already expose the function itself, can't users just call the function and chain it if they need to?

**Comment by mvanderkamp** [comment]
I need to sleep on these changes I think. I've never set up a suite of vimscript unit tests before but it would make changes like this easier if I could point to some unit tests to prove that edge cases are covered.

**Comment by mvanderkamp** [comment]
Okay here's the kind of erroneous call that will trip us up, and I don't see how to resolve this:
```vim
let myvar = '42'
VimuxRunCommand('echo '.myvar)
```
The expression needs to be evaluated in the original context in order to have access to `myvar`. I'm not sure right now how to achieve that. We might be stuck with `<args>` unless we're willing to break a few configs :(.

**Comment by mvanderkamp** [comment]
I'm gonna close this. If someone comes up with a simple robust way of deprecating `<args>` here then I do still think it's worth pursuing, but I haven't dreamt anything up. The execution context issue is a killer from my perspective. Best alternative is to move to new versions of the commands, say `VimuxRun` and `VimuxPrompt` and deprecate the old ones, not sure if it's worth the hassle though.

**Comment by alerque** [comment]
> We might be stuck with `<args>` unless we're willing to break a few configs disappointed.

I'm okay with breaking a few configs as long as we're clearly improving something in the process. What I don't want to do is take something away that people were getting millage out of without offering them a way to at least achieve functional parity.
